### ISSUE TITLE
IPT 2.4.1 regression: cannot open settings

### ISSUE THREAD

**Issue by frafra** [gbif/ipt#1499]
Hi,
I tried to upgrade to IPT 2.4.1 by using the Docker image provided on DockerHub, but when I open `/admin/config.do` I hit an error, which does not appear with version 2.4.0.

```
HTTP Status 500 – Internal Server Error

Type Exception Report

Message Unable to instantiate Action, org.gbif.ipt.action.admin.ConfigAction, defined for 'config' in namespace '/admin'Unable to provision, see the following errors:

Description The server encountered an unexpected condition that prevented it from fulfilling the request.

Exception

Unable to instantiate Action, org.gbif.ipt.action.admin.ConfigAction,  defined for 'config' in namespace '/admin'Unable to provision, see the following errors:

1) Error injecting constructor, com.thoughtworks.xstream.converters.ConversionException: Invalid reference
---- Debugging information ----
message             : Invalid reference
reference           : ../../../../mapping[2]/fields/field/term
referenced-type     : org.gbif.dwc.terms.IucnTerm
referenceable       : false
class               : org.gbif.ipt.model.PropertyMapping
required-type       : org.gbif.ipt.model.PropertyMapping
converter-type      : com.thoughtworks.xstream.converters.reflection.ReflectionConverter
path                : /resource/mappings/mapping[3]/fields/field/term
line number         : 288
class[1]            : java.util.TreeSet
required-type[1]    : java.util.TreeSet
converter-type[1]   : com.thoughtworks.xstream.converters.collections.TreeSetConverter
class[2]            : org.gbif.ipt.model.ExtensionMapping
required-type[2]    : org.gbif.ipt.model.ExtensionMapping
class[3]            : java.util.ArrayList
required-type[3]    : java.util.ArrayList
converter-type[2]   : com.thoughtworks.xstream.converters.collections.CollectionConverter
class[4]            : org.gbif.ipt.model.Resource
required-type[4]    : org.gbif.ipt.model.Resource
version             : 1.4.12
-------------------------------
  at org.gbif.ipt.config.ConfigManagerImpl.<init>(ConfigManagerImpl.java:67)
  at org.gbif.ipt.config.ConfigManagerImpl.class(ConfigManagerImpl.java:67)
  while locating org.gbif.ipt.config.ConfigManagerImpl
  while locating org.gbif.ipt.service.admin.ConfigManager
    for the 4th parameter of org.gbif.ipt.action.admin.ConfigAction.<init>(ConfigAction.java:52)
  while locating org.gbif.ipt.action.admin.ConfigAction

1 error - action - file:/usr/local/tomcat/webapps/ROOT/WEB-INF/classes/struts-admin.xml:25:74
	com.opensymphony.xwork2.DefaultActionInvocation.createAction(DefaultActionInvocation.java:320)
	com.opensymphony.xwork2.DefaultActionInvocation.init(DefaultActionInvocation.java:401)
	com.opensymphony.xwork2.DefaultActionProxy.prepare(DefaultActionProxy.java:201)
	org.apache.struts2.factory.StrutsActionProxy.prepare(StrutsActionProxy.java:57)
	org.apache.struts2.factory.StrutsActionProxyFactory.createActionProxy(StrutsActionProxyFactory.java:32)
	com.opensymphony.xwork2.DefaultActionProxyFactory.createActionProxy(DefaultActionProxyFactory.java:60)
	org.apache.struts2.dispatcher.Dispatcher.serviceAction(Dispatcher.java:564)
	org.apache.struts2.dispatcher.ExecuteOperations.executeAction(ExecuteOperations.java:79)
	org.apache.struts2.dispatcher.filter.StrutsPrepareAndExecuteFilter.doFilter(StrutsPrepareAndExecuteFilter.java:141)
	com.google.inject.servlet.FilterChainInvocation.doFilter(FilterChainInvocation.java:82)
	org.gbif.ipt.struts2.CorsFilter.doFilter(CorsFilter.java:23)
	com.google.inject.servlet.FilterChainInvocation.doFilter(FilterChainInvocation.java:82)
	com.google.inject.servlet.ManagedFilterPipeline.dispatch(ManagedFilterPipeline.java:120)
	com.google.inject.servlet.GuiceFilter.doFilter(GuiceFilter.java:133)

Root Cause

com.google.inject.ProvisionException: Unable to provision, see the following errors:

1) Error injecting constructor, com.thoughtworks.xstream.converters.ConversionException: Invalid reference
---- Debugging information ----
message             : Invalid reference
reference           : ../../../../mapping[2]/fields/field/term
referenced-type     : org.gbif.dwc.terms.IucnTerm
referenceable       : false
class               : org.gbif.ipt.model.PropertyMapping
required-type       : org.gbif.ipt.model.PropertyMapping
converter-type      : com.thoughtworks.xstream.converters.reflection.ReflectionConverter
path                : /resource/mappings/mapping[3]/fields/field/term
line number         : 288
class[1]            : java.util.TreeSet
required-type[1]    : java.util.TreeSet
converter-type[1]   : com.thoughtworks.xstream.converters.collections.TreeSetConverter
class[2]            : org.gbif.ipt.model.ExtensionMapping
required-type[2]    : org.gbif.ipt.model.ExtensionMapping
class[3]            : java.util.ArrayList
required-type[3]    : java.util.ArrayList
converter-type[2]   : com.thoughtworks.xstream.converters.collections.CollectionConverter
class[4]            : org.gbif.ipt.model.Resource
required-type[4]    : org.gbif.ipt.model.Resource
version             : 1.4.12
-------------------------------
  at org.gbif.ipt.config.ConfigManagerImpl.<init>(ConfigManagerImpl.java:67)
  at org.gbif.ipt.config.ConfigManagerImpl.class(ConfigManagerImpl.java:67)
  while locating org.gbif.ipt.config.ConfigManagerImpl
  while locating org.gbif.ipt.service.admin.ConfigManager
    for the 4th parameter of org.gbif.ipt.action.admin.ConfigAction.<init>(ConfigAction.java:52)
  while locating org.gbif.ipt.action.admin.ConfigAction

1 error
	com.google.inject.internal.InternalProvisionException.toProvisionException(InternalProvisionException.java:226)
	com.google.inject.internal.InjectorImpl$1.get(InjectorImpl.java:1053)
	com.google.inject.internal.InjectorImpl.getInstance(InjectorImpl.java:1086)
	com.google.inject.struts2.Struts2Factory.buildBean(Struts2Factory.java:110)
	com.opensymphony.xwork2.ObjectFactory.buildBean(ObjectFactory.java:191)
	com.opensymphony.xwork2.ObjectFactory.buildBean(ObjectFactory.java:177)
	com.opensymphony.xwork2.factory.DefaultActionFactory.buildAction(DefaultActionFactory.java:40)
	com.opensymphony.xwork2.ObjectFactory.buildAction(ObjectFactory.java:142)
	com.opensymphony.xwork2.DefaultActionInvocation.createAction(DefaultActionInvocation.java:301)
	com.opensymphony.xwork2.DefaultActionInvocation.init(DefaultActionInvocation.java:401)
	com.opensymphony.xwork2.DefaultActionProxy.prepare(DefaultActionProxy.java:201)
	org.apache.struts2.factory.StrutsActionProxy.prepare(StrutsActionProxy.java:57)
	org.apache.struts2.factory.StrutsActionProxyFactory.createActionProxy(StrutsActionProxyFactory.java:32)
	com.opensymphony.xwork2.DefaultActionProxyFactory.createActionProxy(DefaultActionProxyFactory.java:60)
	org.apache.struts2.dispatcher.Dispatcher.serviceAction(Dispatcher.java:564)
	org.apache.struts2.dispatcher.ExecuteOperations.executeAction(ExecuteOperations.java:79)
	org.apache.struts2.dispatcher.filter.StrutsPrepareAndExecuteFilter.doFilter(StrutsPrepareAndExecuteFilter.java:141)
	com.google.inject.servlet.FilterChainInvocation.doFilter(FilterChainInvocation.java:82)
	org.gbif.ipt.struts2.CorsFilter.doFilter(CorsFilter.java:23)
	com.google.inject.servlet.FilterChainInvocation.doFilter(FilterChainInvocation.java:82)
	com.google.inject.servlet.ManagedFilterPipeline.dispatch(ManagedFilterPipeline.java:120)
	com.google.inject.servlet.GuiceFilter.doFilter(GuiceFilter.java:133)

Root Cause

com.thoughtworks.xstream.converters.ConversionException: Invalid reference
---- Debugging information ----
message             : Invalid reference
reference           : ../../../../mapping[2]/fields/field/term
referenced-type     : org.gbif.dwc.terms.IucnTerm
referenceable       : false
class               : org.gbif.ipt.model.PropertyMapping
required-type       : org.gbif.ipt.model.PropertyMapping
converter-type      : com.thoughtworks.xstream.converters.reflection.ReflectionConverter
path                : /resource/mappings/mapping[3]/fields/field/term
line number         : 288
class[1]            : java.util.TreeSet
required-type[1]    : java.util.TreeSet
converter-type[1]   : com.thoughtworks.xstream.converters.collections.TreeSetConverter
class[2]            : org.gbif.ipt.model.ExtensionMapping
required-type[2]    : org.gbif.ipt.model.ExtensionMapping
class[3]            : java.util.ArrayList
required-type[3]    : java.util.ArrayList
converter-type[2]   : com.thoughtworks.xstream.converters.collections.CollectionConverter
class[4]            : org.gbif.ipt.model.Resource
required-type[4]    : org.gbif.ipt.model.Resource
version             : 1.4.12
-------------------------------
	com.thoughtworks.xstream.core.AbstractReferenceUnmarshaller.convert(AbstractReferenceUnmarshaller.java:58)
	com.thoughtworks.xstream.core.TreeUnmarshaller.convertAnother(TreeUnmarshaller.java:66)
	com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.unmarshallField(AbstractReflectionConverter.java:499)
	com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.doUnmarshal(AbstractReflectionConverter.java:425)
	com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.unmarshal(AbstractReflectionConverter.java:277)
	com.thoughtworks.xstream.core.TreeUnmarshaller.convert(TreeUnmarshaller.java:72)
	com.thoughtworks.xstream.core.AbstractReferenceUnmarshaller.convert(AbstractReferenceUnmarshaller.java:72)
	com.thoughtworks.xstream.core.TreeUnmarshaller.convertAnother(TreeUnmarshaller.java:66)
	com.thoughtworks.xstream.core.TreeUnmarshaller.convertAnother(TreeUnmarshaller.java:50)
	com.thoughtworks.xstream.converters.collections.AbstractCollectionConverter.readBareItem(AbstractCollectionConverter.java:132)
	com.thoughtworks.xstream.converters.collections.AbstractCollectionConverter.readItem(AbstractCollectionConverter.java:117)
	com.thoughtworks.xstream.converters.collections.TreeSetConverter$1.putCurrentEntryIntoMap(TreeSetConverter.java:153)
	com.thoughtworks.xstream.converters.collections.TreeMapConverter.populateTreeMap(TreeMapConverter.java:113)
	com.thoughtworks.xstream.converters.collections.TreeSetConverter.unmarshal(TreeSetConverter.java:126)
	com.thoughtworks.xstream.core.TreeUnmarshaller.convert(TreeUnmarshaller.java:72)
	com.thoughtworks.xstream.core.AbstractReferenceUnmarshaller.convert(AbstractReferenceUnmarshaller.java:72)
	com.thoughtworks.xstream.core.TreeUnmarshaller.convertAnother(TreeUnmarshaller.java:66)
	com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.unmarshallField(AbstractReflectionConverter.java:499)
	com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.doUnmarshal(AbstractReflectionConverter.java:425)
	com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.unmarshal(AbstractReflectionConverter.java:277)
	com.thoughtworks.xstream.core.TreeUnmarshaller.convert(TreeUnmarshaller.java:72)
	com.thoughtworks.xstream.core.AbstractReferenceUnmarshaller.convert(AbstractReferenceUnmarshaller.java:72)
	com.thoughtworks.xstream.core.TreeUnmarshaller.convertAnother(TreeUnmarshaller.java:66)
	com.thoughtworks.xstream.core.TreeUnmarshaller.convertAnother(TreeUnmarshaller.java:50)
	com.thoughtworks.xstream.converters.collections.AbstractCollectionConverter.readBareItem(AbstractCollectionConverter.java:132)
	com.thoughtworks.xstream.converters.collections.AbstractCollectionConverter.readItem(AbstractCollectionConverter.java:117)
	com.thoughtworks.xstream.converters.collections.CollectionConverter.addCurrentElementToCollection(CollectionConverter.java:98)
	com.thoughtworks.xstream.converters.collections.CollectionConverter.populateCollection(CollectionConverter.java:91)
	com.thoughtworks.xstream.converters.collections.CollectionConverter.populateCollection(CollectionConverter.java:85)
	com.thoughtworks.xstream.converters.collections.CollectionConverter.unmarshal(CollectionConverter.java:80)
	com.thoughtworks.xstream.core.TreeUnmarshaller.convert(TreeUnmarshaller.java:72)
	com.thoughtworks.xstream.core.AbstractReferenceUnmarshaller.convert(AbstractReferenceUnmarshaller.java:72)
	com.thoughtworks.xstream.core.TreeUnmarshaller.convertAnother(TreeUnmarshaller.java:66)
	com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.unmarshallField(AbstractReflectionConverter.java:499)
	com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.doUnmarshal(AbstractReflectionConverter.java:425)
	com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.unmarshal(AbstractReflectionConverter.java:277)
	com.thoughtworks.xstream.core.TreeUnmarshaller.convert(TreeUnmarshaller.java:72)
	com.thoughtworks.xstream.core.AbstractReferenceUnmarshaller.convert(AbstractReferenceUnmarshaller.java:72)
	com.thoughtworks.xstream.core.TreeUnmarshaller.convertAnother(TreeUnmarshaller.java:66)
	com.thoughtworks.xstream.core.TreeUnmarshaller.convertAnother(TreeUnmarshaller.java:50)
	com.thoughtworks.xstream.core.TreeUnmarshaller.start(TreeUnmarshaller.java:134)
	com.thoughtworks.xstream.core.AbstractTreeMarshallingStrategy.unmarshal(AbstractTreeMarshallingStrategy.java:32)
	com.thoughtworks.xstream.XStream.unmarshal(XStream.java:1487)
	com.thoughtworks.xstream.XStream.unmarshal(XStream.java:1467)
	com.thoughtworks.xstream.XStream.fromXML(XStream.java:1347)
	org.gbif.ipt.service.manage.impl.ResourceManagerImpl.loadFromDir(ResourceManagerImpl.java:1037)
	org.gbif.ipt.service.manage.impl.ResourceManagerImpl.loadFromDir(ResourceManagerImpl.java:1023)
	org.gbif.ipt.service.manage.impl.ResourceManagerImpl.load(ResourceManagerImpl.java:984)
	org.gbif.ipt.config.ConfigManagerImpl.loadDataDirConfig(ConfigManagerImpl.java:213)
	org.gbif.ipt.config.ConfigManagerImpl.<init>(ConfigManagerImpl.java:81)
	org.gbif.ipt.config.ConfigManagerImpl$$FastClassByGuice$$7d0dc00c.newInstance(<generated>)
	com.google.inject.internal.DefaultConstructionProxyFactory$FastClassProxy.newInstance(DefaultConstructionProxyFactory.java:89)
	com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:114)
	com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:91)
	com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)
	com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
	com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:148)
	com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:39)
	com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:62)
	com.google.inject.internal.SingleParameterInjector.inject(SingleParameterInjector.java:42)
	com.google.inject.internal.SingleParameterInjector.getAll(SingleParameterInjector.java:65)
	com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:113)
	com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:91)
	com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)
	com.google.inject.internal.InjectorImpl$1.get(InjectorImpl.java:1050)
	com.google.inject.internal.InjectorImpl.getInstance(InjectorImpl.java:1086)
	com.google.inject.struts2.Struts2Factory.buildBean(Struts2Factory.java:110)
	com.opensymphony.xwork2.ObjectFactory.buildBean(ObjectFactory.java:191)
	com.opensymphony.xwork2.ObjectFactory.buildBean(ObjectFactory.java:177)
	com.opensymphony.xwork2.factory.DefaultActionFactory.buildAction(DefaultActionFactory.java:40)
	com.opensymphony.xwork2.ObjectFactory.buildAction(ObjectFactory.java:142)
	com.opensymphony.xwork2.DefaultActionInvocation.createAction(DefaultActionInvocation.java:301)
	com.opensymphony.xwork2.DefaultActionInvocation.init(DefaultActionInvocation.java:401)
	com.opensymphony.xwork2.DefaultActionProxy.prepare(DefaultActionProxy.java:201)
	org.apache.struts2.factory.StrutsActionProxy.prepare(StrutsActionProxy.java:57)
	org.apache.struts2.factory.StrutsActionProxyFactory.createActionProxy(StrutsActionProxyFactory.java:32)
	com.opensymphony.xwork2.DefaultActionProxyFactory.createActionProxy(DefaultActionProxyFactory.java:60)
	org.apache.struts2.dispatcher.Dispatcher.serviceAction(Dispatcher.java:564)
	org.apache.struts2.dispatcher.ExecuteOperations.executeAction(ExecuteOperations.java:79)
	org.apache.struts2.dispatcher.filter.StrutsPrepareAndExecuteFilter.doFilter(StrutsPrepareAndExecuteFilter.java:141)
	com.google.inject.servlet.FilterChainInvocation.doFilter(FilterChainInvocation.java:82)
	org.gbif.ipt.struts2.CorsFilter.doFilter(CorsFilter.java:23)
	com.google.inject.servlet.FilterChainInvocation.doFilter(FilterChainInvocation.java:82)
	com.google.inject.servlet.ManagedFilterPipeline.dispatch(ManagedFilterPipeline.java:120)
	com.google.inject.servlet.GuiceFilter.doFilter(GuiceFilter.java:133)

Note The full stack trace of the root cause is available in the server logs.
Apache Tomcat/8.5.57
```

**Comment by MattBlissett** [comment]
Hi,

This should have been fixed by the 2.4.2 release.  Would you try that please?

Thanks.

**Comment by frafra** [comment]
It has been fixed indeed. Thank you :)
